skbio/stats/composition.py

```diff
@@ -1105,6 +1212,34 @@ def _gram_schmidt_basis(n):
     return basis.T
 
 
+def _build_basis(sbp):
+    """
+    Builds an orthogonal basis from a sequential binary partition. 
+       
+    "Code for a SBP. The first row represents the labels of parts of the compositional vector. At each order of
+    partition, +1 means that the part is assigned to the first group, −1 to the second group, and 0 that this part is not
+    in the group which is divided at this order."
+    J.J. Egozcue and V. Pawlowsky-Glahn, http://dugi-doc.udg.edu/bitstream/handle/10256/660/Egozcue-Pawlowsky.pdf
```

Comment:
Turns out that every SBP can be represented by a tree.

Given the example that you illustrated
```python
sbp = pd.DataFrame(np.array([[1,-1,-1], [0, 1,-1]]), columns=['a', 'b', 'c'])
sbp
   a  b  c
0  1 -1 -1
1  0  1 -1
```
That can also be represented as a tree as follows
```
t = TreeNode.read([r"((b, c), a);"])

print(t.ascii_art())
                    /-b
          /--------|
---------|          \-c
         |
          \-a
```
where `a` constrasts with `b` and `c` in the first vector / branch
and `b` constrasts with `c` in the second vector / branch.
There is more explanation about the orthonormal basis in equation 2 in this [manuscript](http://msystems.asm.org/content/2/1/e00162-16).